jetty-client/src/main/java/org/eclipse/jetty/client/Origin.java

```diff
@@ -63,14 +80,23 @@ public boolean equals(Object obj)
         if (obj == null || getClass() != obj.getClass())
             return false;
         Origin that = (Origin)obj;
-        return scheme.equals(that.scheme) && address.equals(that.address);
+        return scheme.equals(that.scheme) &&
+            address.equals(that.address) &&
+            Objects.equals(tag, that.tag);
     }
 
     @Override
     public int hashCode()
     {
-        int result = scheme.hashCode();
-        result = 31 * result + address.hashCode();
+        return Objects.hash(scheme, address, tag);
```

Comment:
Why no `equals` implementation if you have a `hashCode` impl?

Reply:
Implementation of `equals()` is just above this method.